### ISSUE TITLE
Moving feature flag out of crate root breaks build when using `portable_simd`

### ISSUE THREAD

**Issue by benjamincburns** [VirxEC/rocketsim-rs#6]
This [conditional feature gate](https://github.com/VirxEC/rocketsim-rs/blame/4c89ba2d97c6e762738501aa39e40110535472dd/src/glam_ext.rs#L1) needs to go at the top of `lib.rs`.

Without that change, I get the following error:

```
warning: crate-level attribute should be in the root module
 --> src/glam_ext.rs:1:68
  |
1 | #![cfg_attr(not(any(target_arch = "x86", target_arch = "x86_64")), feature(portable_simd))]
  |                                                                    ^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_attributes)]` on by default

error[E0658]: use of unstable library feature 'portable_simd'
  --> src/glam_ext.rs:10:5
   |
10 | use core::simd::*;
   |     ^^^^^^^^^^
   |
   = note: see issue #86656 <https://github.com/rust-lang/rust/issues/86656> for more information
   = help: add `#![feature(portable_simd)]` to the crate attributes to enable

error[E0658]: use of unstable library feature 'portable_simd'
  --> src/glam_ext.rs:15:14
   |
15 | type F32x4 = f32x4;
   |              ^^^^^
   |
   = note: see issue #86656 <https://github.com/rust-lang/rust/issues/86656> for more information
   = help: add `#![feature(portable_simd)]` to the crate attributes to enable
```

I'd PR the one-line fix, but I'm not in a position to do so at the moment. Dropping an issue as a reminder for myself to do so next time I'm working on `rocketsim-rs` (or just in case @VirxEC wants to get to it before me).

**Comment by benjamincburns** [comment]
Not PRing this right now because the ugly autocxx issue came back to the current `HEAD` of `master`, and I don't want to troubleshoot that first.

**Comment by benjamincburns** [comment]
Side note, I'd imagine this wouldn't have become a problem if #7 wasn't also an issue.

**Comment by VirxEC** [comment]
The reason the current version isn't building is because of a mistake in RocketSim itself, Zealan made a mistake and once he fixes it I'll just need to update to the latest commit. I'll fix this issue too when I do.

**Comment by VirxEC** [comment]
Fixed with commit a11379b
